apps/console/src/features/identity-verification-providers/utils/validation-utils.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { I18n } from "@wso2is/i18n";
+import { FormValidation } from "@wso2is/validation";
+import { IdentityVerificationProviderInterface } from "../models";
+
+export const validateIDVPName = (
+    value: string,
+    idvpList: IdentityVerificationProviderInterface[],
+    currentIDVP?: IdentityVerificationProviderInterface
+): string => {
+
+    if (!value) {
+        return I18n.instance.t("console:develop.features.idvp.forms.generalDetails.name.validations.required");
+    }
+    if (!FormValidation.isValidResourceName(value)) {
+        return I18n.instance.t("console:develop.features.idvp.forms.generalDetails.name.validations.invalid",
+            { idpName: value });
+    }
+
+    if (isIDVPNameAlreadyTaken(value, idvpList, currentIDVP)) {
+        return I18n.instance.t("console:develop.features.idvp.forms.generalDetails.name.validations.duplicate");
+    }
+
+    return;
+};
+
+const isIDVPNameAlreadyTaken = (
```

Comment:
Shall we add TSDoc comments to these functions?

Reply:
Resolved by 84818a49253c03523f196f637f93dc7db9dcdec2